client/main.go

```diff
@@ -0,0 +1,90 @@
+package main
+
+import (
```

Comment:
I want you to change this code to be OOP. 
ii know that foe now that you have one implant for the server and the client. 
but still i want you to use interface.

Reply:
I've created 2 new interfaces - one for the server and one for the client